processing/src/main/java/org/apache/druid/query/filter/FilterBundle.java

```diff
@@ -240,6 +248,21 @@ public MatcherBundleInfo getMatcher()
       return matcher;
     }
 
+    /**
+     * Return a multiline description string, suitable for comparisons in tests.
+     */
+    public String describe()
```

Comment:
toString as well as json serialization is also implemented for all of these as well, is the multi-line nicer for tests i guess?

Reply:
Yeah that's the reason. The new method is used in test cases and makes them easy to read and uodate. I kept the toString the way it was since it's used in log messages, and wasn't sure if it made sense for those to make the log messages multiline.